### PR TITLE
Isupport normalisation

### PR DESCRIPTION
**Pull request by prawnsalad**

### CODE REVIEW

File: src/worker/connectionincoming.js [hunk 1]
```diff
@@ -151,8 +151,40 @@ class ConnectionIncoming {
         this.state.realname = upstream.state.realname;
 
         let nick = this.state.nick;
+
+        // Let plugins modify/add isupport tokens
+        let isupports = upstream.state.registrationLines.filter((regLine) => regLine[0] === '005');
+        let isupportTokens = isupports.reduce((prev, cur) => {
+            // Remove the last token if it contains a space. Usually 'is supported by this server'.
```
Comment (DanielOaks) [inline src/worker/connectionincoming.js]: Not gonna cause any issues, but as a note the space check may mean that if the server outputs a localised translation for _"is supported by this server"_ (that includes no spaces, e.g. an Asian language or something) then we'd interpret the last param as a token. I think every server I've seen outputs the last param as the human-readable description thing.

File: src/worker/connectionincoming.js [hunk 2]
```diff
@@ -279,4 +311,29 @@ class ConnectionIncoming {
     }
 }
 
+// Split []tokens into batches that when joined with prefix and suffix, its length does not exceed maxLen
+// eg. ['one', 'two', ...100 more tokens] = [['one', 'two'], [...more tokens]]
+function batchIsupportTokensToMaxLenth(tokens, prefix, suffix='is supported by this server', maxLen=512) {
+    // 1 = the extra space after the prefix
+    let l = prefix.length + 1;
+    // 1 = the : before the suffix
+    l += suffix.length + 1;
```
Comment (DanielOaks) [inline src/worker/connectionincoming.js]: Should we add an additional 2 for `\r\n`? I noticed below we have a `+1` but not sure whether that's for the space between the `005` and params or for just the `\n` alone.

**Comment by DanielOaks** [comment]
Overall looks pretty good, gonna give it a test and see if I can find any edge cases

**Comment by DanielOaks** [comment]
This should also change [`/src/worker/extensions/chathistory-kiwi/index.js`](https://github.com/kiwiirc/kiwibnc/blob/isupport_normalisation/src/worker/extensions/chathistory-kiwi/index.js) to add the token rather than relying on the `001` sleeping hack.

**Comment by DanielOaks** [comment]
Should we also change [`registerLocalClient()`](https://github.com/kiwiirc/kiwibnc/blob/isupport_normalisation/src/worker/connectionincoming.js#L117) to output its `005` tokens based on the new token list?
